src/components/map.js

```diff
@@ -281,108 +361,239 @@ let Map = class Map extends React.Component {
+      } else {
+        if (this.map.getLayer("clusters")) this.map.removeLayer("clusters");
+        if (this.map.getLayer("cluster-count"))
+          this.map.removeLayer("cluster-count");
+        if (this.map.getLayer("unclustered-point"))
+          this.map.removeLayer("unclustered-point");
+
+          const popup = document.getElementsByClassName('mapboxgl-popup');
+          if ( popup.length ) {
+              popup[0].remove();
+          }
+
+        if (!this.map.getLayer("sa2-fills"))
+          this.map.addLayer({
+            id: "sa2-fills",
+            type: "fill",
+            source: "sa2",
+            sourceLayer: "original",
+            layout: {},
+            paint: {
+              "fill-color": {
+                property: this.props.active.property,
+                stops: this.props.active.stops,
+              },
+              "fill-opacity": [
+                "case",
+                ["boolean", ["feature-state", "click"], false],
+                1,
+                ["boolean", ["feature-state", "highlight"], false],
+                1,
+                ["boolean", ["feature-state", "hover"], false],
+                1,
+                0.8,
+              ],
+            },
+          });
+        let fillColor = {
+          property: this.props.active.property,
+          stops: this.props.active.stops,
+        };
 
-      this.map.setPaintProperty("sa2-fills", "fill-color", fillColor);
+        this.map.setPaintProperty("sa2-fills", "fill-color", fillColor);
+      }
     }
 
     if (this.props.highlightedFeature !== prevProps.highlightedFeature) {
-      this.highlightFeature(this.props.highlightedFeature)
+      this.highlightFeature(this.props.highlightedFeature);
     }
 
     if (this.props.selectedFeature !== prevProps.selectedFeature) {
-      this.selectFeature(this.props.selectedFeature)
+      this.selectFeature(this.props.selectedFeature);
     }
 
-    if (this.props.comparisonFeatures !== prevProps.comparisonFeatures && this.props.mini) {
-      this.highlightComparisonFeatures(this.props.comparisonFeatures)
+    if (
+      this.props.comparisonFeatures !== prevProps.comparisonFeatures &&
+      this.props.mini
+    ) {
+      this.highlightComparisonFeatures(this.props.comparisonFeatures);
     }
   }
 
   selectFeature = (feature) => {
-    const prevId = this.state.selectedFeature?.properties?.SA2_MAIN16
-    const newId = feature?.properties?.SA2_MAIN16
+    const prevId = this.state.selectedFeature?.properties?.SA2_MAIN16;
+    const newId = feature?.properties?.SA2_MAIN16;
     // Skip if the selection hasn't actually changed or if the map isn't fully loaded yet
-    if (prevId === newId || !this.map.getSource("sa2") || !this.map.isSourceLoaded("sa2")) {
-      return
+    if (
+      prevId === newId ||
+      !this.map.getSource("sa2") ||
+      !this.map.isSourceLoaded("sa2")
+    ) {
+      return;
+    } else if (this.props.active.key !== Constants.MAP_TYPE.BUSINESSES) {
```

Comment:
This prevents the selection from being drawn, but it doesn't prevent the highlight from lingering with no way of being dismissed. The mouse handler that normally does that is part of the `sa2-fills` map layer. Rather than deal with any of that right now, let's say that, for now, if someone manages to call `selectFeature()` while on the "Businesses" layer, we'll make the map/legend switch to the "Mobility" layer to force them to use the code that currently works without odd usability issues.

Reply:
made it switch to mobility layer if select feature is called on business layer but it doesn't seem to update the legend. This might be due to store dispatching manually inside the file vs this.props.dispatch but I'm not sure yet

Reply:
It turns out that's due to a bug in DropdownSelect. By using `initialSelectedItem` instead of `selectedItem` on that, it's ignoring changes to the `activeLayer` state. I'll push a fix for that shortly.